src/main/java/org/springframework/integration/aws/outbound/SqsMessageHandler.java

```diff
@@ -75,12 +75,16 @@
 
 
 	public SqsMessageHandler(AmazonSQSAsync amazonSqs) {
-		this(amazonSqs, null);
+		this(amazonSqs, (ResourceIdResolver) null);
 	}
 
 	public SqsMessageHandler(AmazonSQSAsync amazonSqs, ResourceIdResolver resourceIdResolver) {
+		this(amazonSqs, new DynamicQueueUrlDestinationResolver(amazonSqs, resourceIdResolver));
+	}
+
+	public SqsMessageHandler(AmazonSQSAsync amazonSqs, DynamicQueueUrlDestinationResolver destinationResolver) {
```

Comment:
Please, consider to extend it to be based on the `DestinationResolver<String>` interface instead.

Also there should be an `Assert.notNull(destinationResolver, "destinationResolver must not be null");` do not let to inject `null`.

Reply:
Seems I should also add a null check on amazonSqs?

Reply:
Yes, please 😀